File: Unknown/csharp/basics/null.php
<?php
/**
 * Test out null operators in PHP - PHP has a null-coalescing operator but NOT a null-conditional operator
 * User: Andrew Jarombek
 * Date: 12/29/2018
 */

$ageOptional = null;

// Null-coalescing operator
$age = $ageOptional ?? -1;

assert($age == -1);

File: 2019/01-Jan/01-02-csharp-first-impressions/basics/null.swift
#!/usr/bin/swift

// Explore null operators in Swift
// Created by Andy Jarombek on 12/28/2018.

import Foundation

let ageOptional: Int? = nil

// Nil coalesing operator
let age: Int = ageOptional ?? -1

assert(age == -1)

let username: String? = nil

// Nil-conditional operator
let upperUsername = username?.uppercased() ?? "Nil"

assert(upperUsername == "Nil")
